addons/api/addon/utils/sort-results.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+// created_time is not an attribute defined in models per se, but present in
+// all the resources coming from API. It is the default sorting attribute
+const SORT_DEFAULT_ATTRIBUTE = 'created_time';
+const SORT_DIRECTION_ASCENDING = 'asc';
+const SORT_DIRECTION_DESCENDING = 'desc';
+
+const sortFunctions = {
+  string: (a, b) => String(a).localeCompare(String(b)),
+  date: (a, b) => new Date(a).getTime() - new Date(b).getTime(),
+  number: (a, b) => Number(a) - Number(b),
+  boolean: (a, b) => Number(a) - Number(b),
+};
+
+const getSortValue = (sortValue, sortAttribute) => {
```

Comment:
@hashicc this is the first iteration I come with to support `id` sorting when not being within attributes.
Be aware the error message can be improved 😅 also not sure how to intl the message within a util yet.